### PR TITLE
#2029: Completely replace yarn with npm commands throughout the project

### LINKED ISSUE #2029: Completely replace yarn with npm commands throughout the project
**Describe the solution you'd like**
Completely replace yarn with npm commands throughout the project package.json files and any other script files that may rely on yarn.
This includes package.json scripts in all ./packages/* directories and in example apps:
```
./package.json
./packages/ketcher-react/package.json
./packages/ketcher-core/package.json
./packages/ketcher-standalone/package.json
./example/package.json
```
Each package and example need to be able to be built separately with npm and also together at the same time when build script is run from the project root.

Also, at the moment there are also some commands that are yarn-specific, e.g. the project is currently built via yarn foreach plugin: `yarn workspaces foreach -ptvR --from ketcher-react-app run build`
For such commands either npm alternatives or replacements need to be found.

In addition, yarn.lock file should be replaced with package-lock.json and husky scripts should be updated for local development.

As a result - the project should build correctly using only npm for both local development and deployment.

### PR DESCRIPTION
**Pull request by KonstantinEpam23**
closes #2029 #2030 #2031 #2008

### CODE REVIEW

File: demo/package.json [hunk 1]
```diff
@@ -8,9 +8,9 @@
     "@testing-library/user-event": "^13.5.0",
     "@types/jest": "^27.4.1",
     "@types/node": "^16.11.26",
-    "ketcher-core": "workspace:*",
-    "ketcher-react": "workspace:*",
-    "ketcher-standalone": "workspace:*",
```
Comment (KonstantinEpam23) [inline demo/package.json:13]: This is yarn-specific syntax, that is not supported by npm cli

File: packages/ketcher-react/src/constants.ts [hunk 2]
```diff
@@ -17,3 +17,5 @@
 export const KETCHER_INIT_EVENT_NAME = 'ketcher-init'
 
 export const KETCHER_SAVED_SETTINGS_KEY = 'ketcher_editor_saved_settings'
+
+export const TEST = 1
```
Comment (Nitvex) [inline packages/ketcher-react/src/constants.ts:21]: Guess, we need to remove it?

File: DEVNOTES.md [hunk 3]
```diff
@@ -1,9 +1,41 @@
...
+npm install
+```
+*NOTE!* this command should only be run from root directory
+
+Build the project:
+```
+npm run build
+```
+
 ## Prerequisites
```
Comment (Nitvex) [inline DEVNOTES.md:35]: I propose to move `Prerequisites` section to the top of the file

**Comment by AnastasiiaPlyako** [comment]
I think we also can update read.me files in packages/ketcher-core etc

File: packages/ketcher-react/src/constants.ts (hunk 2 above)
Comment (KonstantinEpam23) [reply packages/ketcher-react/src/constants.ts:21]: Ah, good catch! :) Will remove

**Comment by KonstantinEpam23** [comment]
> I think we also can update read.me files in packages/ketcher-core etc

Well, for people who use our packages as npm modules - we don't really care if they use npm or yarn, they should work for both. These changes are mainly for people who work with the repository directly (like us 😄 )

**Comment by KonstantinEpam23** [comment]
Hello @papillot , thank you checking this PR and pointing this out! I believe you are correct about the reason that the error appears b/c the library is not part of the non-dev builds.
Currently ketcher is mostly oriented on using its separate packages ([ketcher-core](https://www.npmjs.com/package/ketcher-core), [ketcher-react](https://www.npmjs.com/package/ketcher-react), etc) and building your own app (e.g. using the /example or /demo directories from the repository)

In the future we may be looking to avoiding using husky for non-dev builds.
Hopefully this clarifies things a bit, thank you for your input again!

**Pull request merged**